posts/brugner/1.sqlserver-docker-m1.md

```diff
@@ -0,0 +1,82 @@
+
+## Install Docker
+
+There are many ways to install Docker. I prefer downloading Docker for Desktop from their [official site](https://www.docker.com/products/docker-desktop/). Remember to choose the **Apple Chip** version.
+
+![Download Docker for Desktop](./assets/download-docker-desktop.png)
+
+After the installation is done, open Docker For Desktop. You'll be welcomed by a home page that shows no containers running.
+
+![Docker For Desktop](./assets/docker-for-desktop-installed.pngg)
+
+## Running Azure SQL Edge on a container
+
+Open your favorite terminal and run the following commands.
+
+**Pull the image from the registry**
+`docker pull mcr.microsoft.com/azure-sql-edge`
+
+**Run the container**
+`docker run --cap-add SYS_PTRACE -e 'ACCEPT_EULA=1' -e 'MSSQL_SA_PASSWORD=Str#ng_Passw#rd' -p 57000:1433 --name azuresqledge -d mcr.microsoft.com/azure-sql-edge`
+
+Important parameters to notice here:
+
+- **ACCEPT_EULA**: accepts the license agreement.
+- **MSSQL_SA_PASSWORD**: sets the password for the sa user.
+- **p**: maps your machine's port to a container's port, in this case 1433 where SQL runs.
+- name: any name you want to use to identify this container.
+
+After running these commands successfully, go to Docker For Desktop again and you'll see the newly created container up and running.
+
+![Container running](./assets/container-running.png)
+
+## Connecting to the server
+
+In order to connect to database server, download [Azure Data Studio](https://learn.microsoft.com/en-us/sql/azure-data-studio/download-azure-data-studio?view=sql-server-ver16) **arm64** version.
+
+Click on New Connection and fill in the details.
+
+![Connecting through Azure Data Studio](./assets/connecting-through-azure-data-studio.png)
+
+- **Connection type**: Microsoft SQL Server
+- **Server**: localhost,57000 (where 57000 is the port you mapped to the container's port)
+- **Authentication type:** SQL Login
+- **User name**: sa
+- **Password**: Str#ng_Passw#rd (the password you set previously)
+- **Encrypt**: False
+- **Trust server certificate**: True
+
+The **connection** has been successfully **established**!
+
+![Azure Data Studio connected successfully](./assets/azure-data-studio-connected-successfully.png)
+
+## Creating a database
+
+Open a new query window and run the following query:
+`-- Create a new database called 'dev_testdb1' -- Connect to the 'master' database to run this snippet USE master GO -- Create the new database if it does not exist already IF NOT EXISTS ( SELECT [name] FROM sys.databases WHERE [name] = N'dev_testdb1' ) CREATE DATABASE dev_testdb1 GO`
```

Comment:
Why all in one line?
```suggestion
```SQL
-- Create a new database called 'dev_testdb1'
-- Connect to the 'master' database to run this snippet
USE master
GO

-- Create the new database if it does not exist already
IF NOT EXISTS (
  SELECT [name] FROM sys.databases WHERE [name] = N'dev_testdb1'
) CREATE DATABASE dev_testdb1
GO
```
```